numpy/core/arrayprint.py

```diff
@@ -287,30 +303,52 @@ def _get_format_function(data, precision, suppress_small, formatter):
         return StructureFormat(format_functions)
 
     dtypeobj = dtype_.type
-    formatdict = _get_formatdict(data, precision, suppress_small, formatter)
     if issubclass(dtypeobj, _nt.bool_):
-        return formatdict['bool']
+        return _lazyget(formatter, 'bool', _boolFormatter)
     elif issubclass(dtypeobj, _nt.integer):
         if issubclass(dtypeobj, _nt.timedelta64):
-            return formatdict['timedelta']
+            return _lazyget(formatter, 'timedelta', TimedeltaFormat, data)
```

Comment:
In fact, could this be written as `formatter.get('timedelta') or TimedeltaFormat(data)`?